Stratis.Bitcoin/AsyncLoop.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Stratis.Bitcoin.Logging;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin
+{
+    public interface IAsyncLoop
+    {
+        string Name { get; }
+
+        Task Run(TimeSpan? repeatEvery = default(TimeSpan?), TimeSpan? startAfter = default(TimeSpan?));
+        Task Run(CancellationToken cancellation, TimeSpan? repeatEvery = default(TimeSpan?), TimeSpan? startAfter = default(TimeSpan?));
+    }
+
+    public class AsyncLoop : IAsyncLoop
```

Comment:
Now that its an instance there is a good case to add the CancellationProvider in the constructor.
And the method CancelleationToken override that.